src/NServiceBus.RabbitMQ/Delays/DelayInfrastructure.cs

```diff
@@ -0,0 +1,111 @@
+﻿namespace NServiceBus.Transport.RabbitMQ
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Text;
+    using global::RabbitMQ.Client;
+
+    static class DelayInfrastructure
+    {
+        const int maxNumberOfBitsToUse = 28;
+        const int maxLevel = maxNumberOfBitsToUse - 1;
+        public const int MaxDelayInSeconds = (1 << maxNumberOfBitsToUse) - 1;
+        public const string DelayHeader = "NServiceBus.Transport.RabbitMQ.DelayInSeconds";
+        public const string DeadLetteredMessageHeader = "x-death";
+        public const string DeliveryExchange = "nsb.delay-delivery";
+
+        public static string LevelName(int level) => $"nsb.delay-level-{level:D2}";
+
+        public static string BindingKey(string address) => $"#.{address}";
+
+        public static void Build(IModel channel)
+        {
+            var bindingKey = "1.#";
+
+            for (var i = maxLevel; i >= 0; i--)
+            {
+                var currentLevel = LevelName(i);
+                var nextLevel = LevelName(i - 1);
+
+                channel.ExchangeDeclare(currentLevel, ExchangeType.Topic, true);
+
+                var arguments = new Dictionary<string, object>();
+                arguments.Add("x-queue-mode", "lazy");
+                arguments.Add("x-message-ttl", Convert.ToInt64(Math.Pow(2, i)) * 1000);
+
+                var deadLetterExchangeName = i > 0 ? nextLevel : DeliveryExchange;
+                arguments.Add("x-dead-letter-exchange", deadLetterExchangeName);
+
+                channel.QueueDeclare(currentLevel, true, false, false, arguments);
+                channel.QueueBind(currentLevel, currentLevel, bindingKey);
+
+                bindingKey = "*." + bindingKey;
+            }
+
+            bindingKey = "0.#";
+
+            for (var i = maxLevel; i >= 1; i--)
+            {
+                var currentLevel = LevelName(i);
+                var nextLevel = LevelName(i - 1);
+
+                channel.ExchangeBind(nextLevel, currentLevel, bindingKey);
+
+                bindingKey = "*." + bindingKey;
+            }
+
+            channel.ExchangeDeclare(DeliveryExchange, ExchangeType.Topic, true);
+            channel.ExchangeBind(DeliveryExchange, LevelName(0), bindingKey);
+        }
+
+        public static void TearDown(IModel channel)
```

Comment:
OK, let's leave as-is for now. It's internal, so not an immediate concern anyway.